src/pages/credentials/useCredentials.tsx

```diff
@@ -22,85 +22,88 @@ export const useCredentials = () => {
-              }}
-              theme="warning"
-            />
-          );
-        case 'APPROVED':
-          if (!dismissed)
-            return (
-              <Banner
-                iconName="check-circle"
-                title={translate('kyb-success')}
-                subtitle={translate('kyb-success-subtitle')}
-                primaryBtnLabel={translate('kyb-success-btn')}
-                primaryBtnAction={onCreateCredential}
-                theme="success"
-                secondaryBtnLabel={translate('kyb-success-dismiss')}
-                secondaryBtnAction={() => {
-                  localStorage.setItem('dismissed-alert', 'true');
-                  setDismissed(false);
-                }}
-              />
-            );
-          return <></>;
-      }
+  const bannerData: Record<
+    'null' | 'REJECTED' | 'PENDING' | 'APPROVED',
+    {
+      iconName: string;
+      title: string;
+      subtitle: string;
+      primaryBtnLabel: string;
+      primaryBtnAction: () => void;
+      secondaryBtnLabel?: string;
+      secondaryBtnAction?: () => void;
+      theme: 'success' | 'warning' | 'error';
+    }
+  > = {
+    null: {
+      iconName: 'alert-circle',
+      title: translate('kyb-verify'),
+      subtitle: translate('kyb-verify-subtitle'),
+      primaryBtnLabel: translate('kyb-verify-btn'),
+      primaryBtnAction: () => {
+        setOpenModal({ name: 'verify', open: true });
+      },
+      theme: 'warning',
+    },
+    REJECTED: {
+      iconName: 'alert-circle',
+      title: translate('kyb-failed'),
+      subtitle: translate('kyb-failed-subtitle'),
+      primaryBtnLabel: translate('kyb-failed-label'),
+      primaryBtnAction: () => {
+        setOpenModal({ name: 'rejected', open: true });
+      },
+      theme: 'error',
+    },
+    PENDING: {
+      iconName: 'alert-circle',
+      title: translate('kyb-pending'),
+      subtitle: translate('kyb-pending-subtitle'),
+      primaryBtnLabel: translate('kyb-pending-btn'),
+      primaryBtnAction: () => {
+        setOpenModal({ name: 'pending', open: true });
+      },
+      theme: 'warning',
+    },
+    APPROVED: {
+      iconName: 'check-circle',
+      title: translate('kyb-success'),
+      subtitle: translate('kyb-success-subtitle'),
+      primaryBtnLabel: translate('kyb-success-btn'),
+      primaryBtnAction: onCreateCredential,
+      secondaryBtnLabel: translate('kyb-success-dismiss'),
+      secondaryBtnAction: () => {
+        localStorage.setItem('dismissed-alert', 'true');
+        setDismissed(false);
+      },
+      theme: 'success',
+    },
   };
 
   const getIssuedListComponent = () => {
-    if (isVerified)
-      return (
-        <div className="flex flex-col gap-6">
-          {displayBanner()}
-          <IssuedList />
-        </div>
-      );
-    return displayBanner();
+    const status = verificationStatus || 'null';
+    return (
+      <div className="flex flex-col gap-6">
```

Comment:
it's not logic side and it's better to create a container for it and call it here